src/libvirtApi/domain.js

```diff
@@ -506,66 +481,49 @@ export function domainDesktopConsole({
     });
 }
 
-export function domainDetachDisk({
+export async function domainDetachDisk({
     name,
     connectionName,
     id: vmPath,
     target,
     live = false,
     persistent
 }) {
-    let diskXML;
     let detachFlags = Enum.VIR_DOMAIN_AFFECT_CURRENT;
     if (live)
         detachFlags |= Enum.VIR_DOMAIN_AFFECT_LIVE;
 
-    return call(connectionName, vmPath, 'org.libvirt.Domain', 'GetXMLDesc', [0], { timeout, type: 'u' })
-            .then(domXml => {
-                const getXMLFlags = Enum.VIR_DOMAIN_XML_INACTIVE;
-                diskXML = getDiskElemByTarget(domXml[0], target);
+    const [domXml] = await call(connectionName, vmPath, 'org.libvirt.Domain', 'GetXMLDesc', [0], { timeout, type: 'u' });
+    const diskXML = getDiskElemByTarget(domXml, target);
 
-                return call(connectionName, vmPath, 'org.libvirt.Domain', 'GetXMLDesc', [getXMLFlags], { timeout, type: 'u' });
-            })
-            .then(domInactiveXml => {
-                const diskInactiveXML = getDiskElemByTarget(domInactiveXml[0], target);
-                if (diskInactiveXML && persistent)
-                    detachFlags |= Enum.VIR_DOMAIN_AFFECT_CONFIG;
+    const [domInactiveXml] = await call(connectionName, vmPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' });
 
-                return call(connectionName, vmPath, 'org.libvirt.Domain', 'DetachDevice', [diskXML, detachFlags], { timeout, type: 'su' });
-            });
+    const diskInactiveXML = getDiskElemByTarget(domInactiveXml, target);
+    if (diskInactiveXML && persistent)
+        detachFlags |= Enum.VIR_DOMAIN_AFFECT_CONFIG;
+
+    await call(connectionName, vmPath, 'org.libvirt.Domain', 'DetachDevice', [diskXML, detachFlags], { timeout, type: 'su' });
 }
 
 // Cannot use virt-xml until https://github.com/virt-manager/virt-manager/issues/357 is fixed
-export function domainDetachHostDevice({ connectionName, vmId, live, dev }) {
+export async function domainDetachHostDevice({ connectionName, vmId, live, dev }) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1397-20240122-143818-61fe453f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)